sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/DotCoverReportParser.java

```diff
@@ -93,27 +96,39 @@ private String extractFileCanonicalPath(String contents) {
     }
 
     private void collectCoverage(String fileCanonicalPath, String contents) {
-      Matcher matcher = COVERED_LINES_PATTERN_1.matcher(contents);
-      checkMatches(matcher);
-      String highlightedContents = matcher.group(1);
+      Matcher fileCoverageMatcher = FILE_COVERAGE_PATTERN.matcher(contents);
+      if (!fileCoverageMatcher.matches()) {
+        throw new IllegalArgumentException("The report contents does not match the following regular expression: " + FILE_COVERAGE_PATTERN.pattern());
+      }
 
-      matcher = COVERED_LINES_PATTERN_2.matcher(highlightedContents);
+      String highlightedContents = fileCoverageMatcher.group(SEQUENCE_POINTS_GROUP_NAME);
+      Matcher sequencePointsMatcher = SEQUENCE_POINT_PATTERN.matcher(highlightedContents);
 
-      while (matcher.find()) {
-        int lineStart = Integer.parseInt(matcher.group(1));
-        int hits = Integer.parseInt(matcher.group(3));
+      while (sequencePointsMatcher.find()) {
+        int lineStart = Integer.parseInt(sequencePointsMatcher.group(1));
+        int hits = Integer.parseInt(sequencePointsMatcher.group(3));
 
         coverage.addHits(fileCanonicalPath, lineStart, hits);
 
         LOG.trace("dotCover parser: found coverage for line '{}', hits '{}' when analyzing the path '{}'.",
-            lineStart, hits, fileCanonicalPath);
+          lineStart, hits, fileCanonicalPath);
+      }
+    }
+
+    private int getIndexOf(String fileContent, String tag) {
```

Comment:
We could use `public int indexOf(String str, int fromIndex)` overload.
It will start from 0 for the `<title>`
It will start from `<title>` to search for `</title>`, ensuring we have consistent state.

Like this, we can remove `getTitlePath`, as the check will be naturally embedded in the system.
```suggestion
    private int getIndexOf(String fileContent, String tag, int startIndex = 0) {
```

Reply:
I meant it like this to respect previous comment about `getIndexOf`.
```
    private int getIndexOf(String fileContent, String tag, int startIndex) {
      int index = fileContent.indexOf(tag, startIndex);
      if (index == -1) {
        throw new IllegalArgumentException(String.format("The report does not contain a '%s' tag.", tag));
      }
      return index;
    }
```
and usage like
```
      int indexOfTitleStart = getIndexOf(contents, TITLE_START);
      int indexOfTitleEnd = getIndexOf(contents, "</title>", indexOfTitleStart);
      String lowerCaseAbsolutePath = fileContent.substring(indexOfTitleStart + TITLE_START.length(), indexOfTitleEnd);

```
We won't need so many constants as well, as stuff would not be repeated.

Reply:
The </title><title> is such a corner case, that we can live with current error message. We could also update it to something like hardcoded `..not contain valid <title> tag..`

Reply:
(sorry, I misread your comment initially, makes sense)